zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/FlowNodeInstanceQueryController.java

```diff
@@ -45,6 +51,33 @@ public ResponseEntity<FlowNodeInstanceSearchQueryResponse> searchFlownodeInstanc
         .fold(RestErrorMapper::mapProblemToResponse, this::search);
   }
 
+  @GetMapping(
+      path = "/{flowNodeInstanceKey}",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE})
+  public ResponseEntity<FlowNodeInstanceItem> getByKey(
+      @PathVariable("flowNodeInstanceKey") final Long flowNodeInstanceKey) {
+    try {
+      return ResponseEntity.ok()
+          .contentType(new MediaType(MediaType.APPLICATION_JSON, StandardCharsets.UTF_8))
+          .body(
+              SearchQueryResponseMapper.toFlowNodeInstance(
+                  flownodeInstanceServices.getByKey(flowNodeInstanceKey)));
+    } catch (final NotFoundException nfe) {
+      final var problemDetail =
+          RestErrorMapper.createProblemDetail(
+              HttpStatus.NOT_FOUND, nfe.getMessage(), RejectionType.NOT_FOUND.name());
+      return RestErrorMapper.mapProblemToResponse(problemDetail);
+    } catch (final Exception e) {
+      REST_LOGGER.warn("An exception occurred in get flow node instance by key.", e);
+      final var problemDetail =
+          RestErrorMapper.createProblemDetail(
+              HttpStatus.INTERNAL_SERVER_ERROR,
+              e.getMessage(),
+              "Failed to execute Get Flow node instance by key.");
+      return RestErrorMapper.mapProblemToResponse(problemDetail);
+    }
```

Comment:
See [comment](https://github.com/camunda/camunda/pull/22480/files#r1768043590) in `Get task by key`. We should implement this once in one of the open PRs and then reuse it in all GET BY KEY endpoints (can be a follow-up issue/PR, of course).